### ISSUE TITLE
Unable to POST a new Game: 404 Not Found

### ISSUE THREAD

**Issue by kmg1434** [ga-wdi-boston/full-stack-project#528]
So I am attempting to create a new Game. User's have many Games. A Game has an id, user_id foreign key, lvl (integer), and a score (decimal). Upon entering this curl request:
```
curl --include --request POST http://localhost:4741/games \
  --header "Content-Type: application/json" \
  --data '{
    “game”: {
      “user_id": 2,
      “lvl”: 4,
      “score”: 0.1
    }
  }'
```
I get a shit ton of HTML trash back in the shell, the console displays this:

```
jquery.js:9536 POST http://localhost:4741/games 404 (Not Found)
jquery.js:9536 XHR finished loading: POST "http://localhost:4741/games".
error!  // this is a console log from my ui.createGameFailure()
```

I would understand what to do if this were a Bad Request error, and I would check out my jSon formatting, but what exactly is "Not Found"? My url and table exist, I can see the data table at http://localhost:4741/games just fine. My back end server says that it is routing correctly but there is a parse error:

```
Started POST "/games" for ::1 at 2016-12-01 18:01:51 -0500
Error occurred while parsing request parameters.
Contents:

{
    “game”: {
      “user_id": 2,
      “lvl”: 4,
      “score”: 0.1
    }
  }

ActionDispatch::ParamsParser::ParseError (822: unexpected token at '{
    “game”: {
      “user_id": 2,
      “lvl”: 4,
      “score”: 0.1
    }
  }'):
```

Not sure where to go from here. My POST attempt from the front end does the same exact error.

**Comment by raq929** [comment]
Looks like you are sending improperly formatted JSON.  For some reason it seems like you're using a mix of angled and straight quotes around your strings, and the JSON linter I used is NOT HAPPY WITH IT at all. http://jsonlint.com/
I would start a new file (because sometimes weird data things can happen with curl requests) and rewrite the request, making sure to use straight quotes all the way through.
But do it tomorrow, because if you're anything like me, your ability to stare at JSON and fix it drops dramatically at 10pm. 😉

**Comment by neugierige** [comment]
If you are still having trouble, maybe what you can do is create a game object in the rails console, then use a curl GET to see what the JSON looks like...

**Comment by gaand** [comment]
What does the games controller inherit from?

**Comment by kmg1434** [comment]
Ok, so curl request now creates a game! Woo! So I did a lot of things... umm... where do I start.

Here's my curl request that works:

```
API="http://localhost:4741"
URL_PATH="/games"
TOKEN="BAhJIiU3NGZmMzBhNTgyOTYzYmE2MjMwOGNjZGIzN2U3MWEzNgY6BkVG--22e12cff6005b394da18495004dc774380c817df"
curl "${API}${URL_PATH}" \
  --include \
  --request POST \
  --header "Content-Type: application/json" \
  --header "Authorization: Token token=$TOKEN" \
  --data '{"game":{"lvl":2,"score":null}}'
```

When I scaffolded Games, it inherited from ApplicationController, rather than OpenReadController. OpenReadController has a ```before_action``` that calls ```set_current_user```. I also needed to add ```@game = current_user.games.build(game_params)``` to my create function in my GamesController, rather than the stock function the rails scaffold gave me. I found this line in the examples_controller.rb file.

Most of this process consisted of looking at the rails server response after each little change and comparing my code to the example_controller. I know there were a couple more things that I did but honestly they've already escaped me.
